core/App/screens/ProofRequestDetails.tsx

```diff
@@ -0,0 +1,344 @@
+        description: template.description,
+        language: i18n.language,
+      }
+      setMeta(metaOverlay)
+      setAttributes(attributes)
+    })
+  }, [templateId])
+
+  const onlyNumberRegex = /^\d+$/
+
+  const onChangeValue = useCallback(
+    (schema: string, label: string, name: string, value: string) => {
+      if (!onlyNumberRegex.test(value)) {
+        setInvalidPredicate({ visible: true, predicate: label })
+        return
+      }
+      setInvalidPredicate(undefined)
+      setCustomPredicateValues((prev) => ({
+        ...prev,
+        [schema]: {
+          ...(prev[schema] || {}),
+          [name]: parseInt(value),
+        },
+      }))
+    },
+    [setCustomPredicateValues, setInvalidPredicate]
+  )
+
+  const useProofRequest = useCallback(async () => {
+    if (invalidPredicate) {
+      setInvalidPredicate({ visible: true, predicate: invalidPredicate.predicate })
+      return
+    }
+
+    if (connectionId) {
+      // Send to specific contact and redirect to the chat with him
+      sendProofRequest(agent, templateId, connectionId, customPredicateValues).then((result) => {
+        if (result?.proofRecord) {
+          linkProofWithTemplate(agent, result.proofRecord, templateId)
+        }
+      })
+
+      navigation.getParent()?.navigate(Screens.Chat, { connectionId })
+    } else {
+      // Else redirect to the screen with connectionless request
+      navigation.navigate(Screens.ProofRequesting, { templateId, predicateValues: customPredicateValues })
+    }
+  }, [agent, templateId, connectionId, customPredicateValues, invalidPredicate])
+
+  const showTemplateUsageHistory = useCallback(async () => {
+    navigation.navigate(Screens.ProofRequestUsageHistory, { templateId })
+  }, [navigation, templateId])
+
+  const Header: React.FC = () => {
+    return (
+      <View style={style.header}>
+        <Text style={style.title}>{meta?.name}</Text>
+        <Text style={style.description}>{meta?.description}</Text>
+      </View>
+    )
+  }
+
+  const Footer: React.FC = () => {
+    return (
+      <View>
+        <View style={style.footerButton}>
+          <Button
+            title={t('Verifier.ShowTemplateUsageHistory')}
+            accessibilityLabel={t('Verifier.ShowTemplateUsageHistory')}
+            testID={testIdWithKey('ShowTemplateUsageHistory')}
+            buttonType={ButtonType.Secondary}
+            onPress={() => showTemplateUsageHistory()}
+          />
+        </View>
+        <View style={style.footerButton}>
+          <Button
+            title={connectionId ? t('Verifier.SendThisProofRequest') : t('Verifier.UseProofRequest')}
+            accessibilityLabel={connectionId ? t('Verifier.SendThisProofRequest') : t('Verifier.UseProofRequest')}
+            testID={connectionId ? testIdWithKey('SendThisProofRequest') : testIdWithKey('UseProofRequest')}
+            buttonType={ButtonType.Primary}
+            onPress={() => useProofRequest()}
+          />
+        </View>
```

Comment:
Should the primary button be above the secondary button ?

Reply:
Swapped